### ISSUE TITLE
Astroquery appears to have issues with AWS sagemaker

### ISSUE THREAD

**Issue by prmaynard** [astropy/astroquery#1736]
Hello,
I am currently working on a us-east sagemaker jupyternotebook. Unfortunately I keep running into the same error,

The following throws the error below,

`from astroquery.mast import Observations`

---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-17-cb5bb4d21530> in <module>
----> 1 from astroquery.mast import Observations

/opt/conda/lib/python3.7/site-packages/astroquery/mast/__init__.py in <module>
     31 conf = Conf()
     32 
---> 33 from .cutouts import TesscutClass, Tesscut
     34 from .observations import Observations, ObservationsClass, MastClass, Mast
     35 from .collections import Catalogs, CatalogsClass

/opt/conda/lib/python3.7/site-packages/astroquery/mast/cutouts.py in <module>
    315 
    316 
--> 317 Tesscut = TesscutClass()

/opt/conda/lib/python3.7/site-packages/astroquery/mast/cutouts.py in __init__(self)
    102     def __init__(self):
    103 
--> 104         super().__init__()
    105 
    106         services = {"sector": {"path": "sector"},

/opt/conda/lib/python3.7/site-packages/astroquery/mast/core.py in __init__(self, mast_token)
     31     def __init__(self, mast_token=None):
     32 
---> 33         super().__init__()
     34 
     35         # Initializing API connections

/opt/conda/lib/python3.7/site-packages/astroquery/query.py in __init__(self)
    417 
    418     def __init__(self):
--> 419         super(QueryWithLogin, self).__init__()
    420         self._authenticated = False
    421 

/opt/conda/lib/python3.7/site-packages/astroquery/query.py in __init__(self)
    160 
    161         self.cache_location = os.path.join(
--> 162             paths.get_cache_dir(), 'astroquery',
    163             self.__class__.__name__.split("Class")[0])
    164         if not os.path.exists(self.cache_location):

/opt/conda/lib/python3.7/site-packages/astropy/config/paths.py in get_cache_dir(rootname)
    171                 linkto = xchpth
    172 
--> 173     return os.path.abspath(_find_or_create_root_dir('cache', linkto, rootname))
    174 
    175 

/opt/conda/lib/python3.7/site-packages/astropy/config/paths.py in _find_or_create_root_dir(dirnm, linkto, pkgname)
    289 
    290 def _find_or_create_root_dir(dirnm, linkto, pkgname='astropy'):
--> 291     innerdir = os.path.join(_find_home(), '.{}'.format(pkgname))
    292     maindir = os.path.join(_find_home(), '.{}'.format(pkgname), dirnm)
    293 

/opt/conda/lib/python3.7/site-packages/astropy/config/paths.py in _find_home()
     34             homedir = os.environ['HOME']
     35         else:
---> 36             raise OSError('Could not find unix home directory to search for '
     37                           'astropy config dir')
     38     elif os.name == 'nt':  # This is for all modern Windows (NT or after)

OSError: Could not find unix home directory to search for astropy config dir

However, there seems to be no issue with:
`import astroquery`

Additionally when I try,

`from astropy.config import get_config_dirs`

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-19-3971ca575119> in <module>
----> 1 from astropy.config import get_config_dirs

ImportError: cannot import name 'get_config_dirs' from 'astropy.config' (/opt/conda/lib/python3.7/site-packages/astropy/config/__init__.py)

Any help would be greatly appreciated thanks.

**Comment by keflavich** [comment]
Well, that's pretty interesting!  No home directory! 

I think this is a more generic astropy issue, though.  @ceb8 , do you have experience w/this from setting up amazon things for MAST?

**Comment by ceb8** [comment]
Hmmmmm, that is interesting. I haven't seen this, but I've only ever been working on platforms where the user does have a home directory. Maybe @mustaric has experiences this.

**Comment by mustaric** [comment]
I once wrote a AWS lambda function which similarly didn't have a home directory.  I didn't need any of the information that astropy was writing to that directory because I was just using the fits.io.  So I did the following in my import statements:

import warnings
from astropy.utils.data import CacheMissingWarning
from astropy.config import ConfigurationMissingWarning
from astropy.config import InvalidConfigurationItemWarning
warnings.simplefilter('ignore', CacheMissingWarning)
warnings.simplefilter('ignore', ConfigurationMissingWarning)
warnings.simplefilter('ignore', InvalidConfigurationItemWarning)
from astropy.io import fits

I worry there was also something about setting up an environmental variable to redirect the output to a /tmp directory, but I can't find it in my notes.

I hope this helps.

**Comment by mustaric** [comment]
This page might be of some use.
https://docs.astropy.org/en/stable/config/
set the XDG_CONFIG_HOME and XDG_CACHE_HOME to something else, if that is possible.

**Comment by prmaynard** [comment]
So I can import get_config_dir but when I run,

`get_config_dir()`

I get back another error message saying it can't find my home directory 

![image](https://user-images.githubusercontent.com/55207569/84176517-82b10900-aa4f-11ea-8a64-0b4f6fe2d88c.png)

Admittedly I don't know much about file systems so please forgive any unintended ignorance I have working with UNIX. Here is a bit of the directories,

![image](https://user-images.githubusercontent.com/55207569/84177001-3ca87500-aa50-11ea-92d1-8f5e273dac8f.png)

I am not sure if this is the home directory in question or not, I can show more if it would help. I am trying to work with some of the uncalibrated and calibrated FFI's and Astroquery with the AWS bucket seems to be the best way.

@mustaric unfortunately the imports and ignore warnings didn't help, it still throws the same error message. Please let me know if there is any more information I can provide that would be helpful. Thanks for the efforts so far.

Also unfortunately with my UNIX inexperience I am not exactly sure how I would set XDG_CONFIG_HOME and XDG_CACHE_HOME to something else.

**Comment by prmaynard** [comment]
**The resolution for future reference**

Never mind, as it turns out and as @mustaric pointed out, the section from the astropy documentation is prepared for exactly this situation where the home directory is not accessible. For anyone who might come across this in the future I am using amazon sagemaker studio and this is how I was able to use astropy:

1. Create a directory to store the libraries, in this case I named it 'HOME'

2. Use this to clone astropy and then put it in the directory from step one
`!git clone --recursive git://github.com/astropy/astropy.git`

3. Next use the following to point XDG_CONFIG_HOME and XDG_CACHE_HOME to the newly created directory (for me this is named 'HOME')

`import os`
`os.environ['XDG_CONFIG_HOME'] = 'HOME'`
`os.environ['XDG_CACHE_HOME'] = 'HOME'`

4. Running this:
`from astropy.config import get_config_dir`
should yield,
`'/root/HOME/astropy'`


After this I had no problems and it works same as normal, Thanks!
